### PR TITLE
Bugfix 1111 item only

### PR DESCRIPTION
**Pull request by wdednam**
This change is [Reviewable](https://reviewable.io/reviews/highperformancecoder/minsky/133)

### CODE REVIEW

File: test/00/groupResize.sh [hunk 1]
```diff
@@ -46,8 +46,9 @@ proc afterMinskyStarted {} {uplevel #0 {
  canvas.mouseDown [expr \$x+0.5*\$w] [expr \$y+0.5*\$h]
  canvas.mouseUp [expr \$x+\$w]  [expr \$y+\$h]
 
- assert "abs([expr 1.5*\$w]-[minsky.canvas.item.width])<5"
- assert "abs([expr 1.5*\$h]-[minsky.canvas.item.height])<5"
+ # Resize handles larger on groups, see ticket 1023, so needed to adjust the factors below
```
Comment (highperformancecoder) [inline test/00/groupResize.sh]: Yes - this. I'm still unhappy with the GroupItem placement of resize handles.

**Comment by highperformancecoder** [comment]
Given a) that I think there is still an issue with Group rendering, and b) that I think I have a better alternative to ecolab::Accessor that we can use in other cases as they arise, I might just hold off merging this until we've looked at those issues.

**Comment by highperformancecoder** [comment]
Sorry - didn't mean to close this just yet.

File: test/00/groupResize.sh (hunk 1 above)
Comment (wdednam) [reply test/00/groupResize.sh]: I realised after I requested the merge that I still hadn't fixed up the test, sorry. Once you have implemented the alternative to the Accessor, I will fix it if still required. I think the problem is caused by a combination of the larger resize handles and the left and right margins of the IO variables, which I still have to figure out how to call in the test TCL code, but haven't yet managed to.

**Comment by highperformancecoder** [comment]
On Sun, Jan 26, 2020 at 02:55:43AM -0800, Wynand wrote:
> @wdednam commented on this pull request.
> 
> ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
> 
> In test/00/groupResize.sh:
> 
> > @@ -46,8 +46,9 @@ proc afterMinskyStarted {} {uplevel #0 {
>   canvas.mouseDown [expr \$x+0.5*\$w] [expr \$y+0.5*\$h]
>   canvas.mouseUp [expr \$x+\$w]  [expr \$y+\$h]
> 
> - assert "abs([expr 1.5*\$w]-[minsky.canvas.item.width])<5"
> - assert "abs([expr 1.5*\$h]-[minsky.canvas.item.height])<5"
> + # Resize handles larger on groups, see ticket 1023, so needed to adjust the factors below
> 
> I realised after I requested the merge that I still hadn't fixed up the test,
> sorry. Once you have implemented the alternative to the Accessor, I will fix it
> if still required. I think the problem is caused by a combination of the larger
> resize handles and the left and right margins of the IO variables, which I
> still have to figure out how to call in the test TCL code, but haven't yet
> managed to.

My theory is that a bb.update() needs to be called after io vars have
been changed, or similar. Also, the way width/height is done in Groups
differs from the other items, and maybe needs to be brought into line
with the other Item types.

-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     hpcoder@hpcoders.com.au
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Comment by highperformancecoder** [comment]
I discovered the solution for the group resize issue. In schema2, bb
needs to be updated on loading the iconWidth/Height parameters. I
changed the name of these for good measure. Also fixed up a few other
places it makes sense to update the bounding box.


-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     hpcoder@hpcoders.com.au
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Comment by wdednam** [comment]
Awesome, thank you. The change you made in schema2.cc works like a charm. The test groupResize.sh works in its original form (no need to change the 1.5 coefficients to 1.3 and 1.4 in the asserts or to include the values of the margins either).

**Comment by highperformancecoder** [comment]
This has now been obsoleted by other developments

**Pull request closed without merging**